frontend/src/metabase/visualizations/echarts/cartesian/option/trend-line.ts

```diff
@@ -0,0 +1,195 @@
+import d3 from "d3";
+// eslint-disable-next-line no-restricted-imports -- deprecated usage
+import moment from "moment-timezone";
+import _ from "underscore";
+import Color from "color";
+import type { RegisteredSeriesOption, EChartsOption } from "echarts";
+
+import { getTrendDataPointsFromInsight } from "metabase/visualizations/lib/trends";
+import type {
+  ComputedVisualizationSettings,
+  RenderingContext,
+} from "metabase/visualizations/types";
+import type { RowValue, SeriesSettings } from "metabase-types/api";
+import type { Insight } from "metabase-types/api/insight";
+
+import { applySquareRootScaling, replaceValues } from "../model/dataset";
+import type { CartesianChartModel, DataKey } from "../model/types";
+
+const TREND_LINE_DATA_KEY = "trend-line";
+
+type TrendDataset = {
+  [key: DataKey]: RowValue;
+  [TREND_LINE_DATA_KEY]: number;
+}[];
+
+/**
+ * Computes the dataset for a single series, based on its `insight` object.
+ *
+ * @param {Insight} insight - Insight object for a series
+ * @param {CartesianChartModel} chartModel - Locally computed chart data
+ * @returns {TrendDataset} Resultant dataset for the series with the given `insight` object
+ */
+function getSingleSeriesTrendDataset(
+  insight: Insight,
+  chartModel: CartesianChartModel,
+): TrendDataset {
+  const xValues = chartModel.dataset.map(
+    // We know the value is a string because it has to be a timeseries aggregation
+    row => moment(row[chartModel.dimensionModel.dataKey] as string),
```

Comment:
AFAIR we decided to use dayjs in the new code. I've just started using it in my very recent PRs and it is a drop-in replacement so I think it makes sense to change it here too